cmd/activator/main.go

```diff
@@ -199,20 +206,21 @@ func main() {
 	params := queue.BreakerParams{QueueDepth: breakerQueueDepth, MaxConcurrency: breakerMaxConcurrency, InitialCapacity: 0}
 
 	// Return the number of endpoints, 0 if no endpoints are found.
-	endpointsGetter := func(rev *v1alpha1.Revision) (int32, error) {
-		// We have to read the private service endpoints in activator
-		// in order to count the serving pod count.
-		// TODO(vagababov): reduce the layer jump and use SKS.Status.PrivateService here.
-		sn := names.PrivateService(rev.Name)
-		count, err := resources.FetchReadyAddressCount(endpointInformer.Lister(), rev.Namespace, sn)
+	endpointsGetter := func(ns, n string) (int32, error) {
+		count, err := resources.FetchReadyAddressCount(endpointInformer.Lister(), ns, n)
 		return int32(count), err
 	}
 
-	// Return the revision from the observer.
+	// Returns the revision from the observer.
 	revisionGetter := func(revID activator.RevisionID) (*v1alpha1.Revision, error) {
```

Comment:
Should this now also take (ns, n string)?

Reply:
No, I kept it as is because it expresses the intent of getting a revision.